ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -62,6 +62,10 @@ pub(crate) struct TraceBuilder {
     // Current count of recursive calls to the function in which outlining was started. Will be 0
     // if `outline_target_blk` is None.
     recursion_count: usize,
+    // Values promoted to constants during runtime. This is only needed during trace building and
+    // can be thrown away after.
+    promotions: Box<[usize]>,
+    promote_idx: usize,
```

Comment:
Can we document this field? [I think I know what it does, but I'll forget in the future!]

Reply:
Sure, fixed here (be9ae27668b8ae05209c1209974434ec35c2dbf0)

Reply:
It's not the "value" is it? I thought it was "what position we are in the promotions array" or similar?

Reply:
How about this? 44851aa4db32ec9a076ec9bbef23ec71f5a9c35b